draft-ietf-quic-transport.md

```diff
@@ -808,27 +809,32 @@ direction until that direction enters a terminal state, or until one of the
 endpoints sends CONNECTION_CLOSE.
 
 
-## Stream Final Offset {#final-offset}
+## Stream Final Size {#final-size}
 
-The final offset is the count of the number of bytes that are transmitted on a
-stream.  For a stream that is reset, the final offset is carried explicitly in a
-RESET_STREAM frame.  Otherwise, the final offset is the offset of the end of the
-data carried in a STREAM frame marked with a FIN flag, or 0 in the case of
-incoming unidirectional streams.
+The final size is the amount of flow control credit that is consumed by a
+stream.  Assuming that every contiguous byte on the stream was sent, the final
+size is the number of bytes sent.  More generally, this is one higher than the
+largest byte offset sent on the stream.
 
-An endpoint will know the final offset for a stream when the receiving part of
-the stream enters the "Size Known" or "Reset Recvd" state ({{stream-states}}).
+For a
+stream that is reset, the final size is carried explicitly in a RESET_STREAM
+frame.  Otherwise, the final size is the offset plus the length of a STREAM
```

Comment:
It might be worth adding that for a stream that is reset, the final size might be smaller than the number of bytes sent on the stream.

Reply:
Isn't that a protocol violation triggering FINAL_OFFSET_ERROR (to be renamed to FINAL_SIZE_ERROR)? Quoting from the draft: _an endpoint received a RESET_STREAM frame containing a final offset that was lower than the maximum offset of data that was already received._

Reply:
Yes, my suggestion was incorrect. Not sure what I was thinking.